compiler/one-cmds/tests/one_codegen_008.py

```diff
@@ -0,0 +1,13 @@
+from onelib import argumentparse
+from onelib.argumentparse import DriverName, NormalOption, TargetOption
+
+
+def command_schema():
+    parser = argumentparse.ArgumentParser()
+    parser.add_argument("dummy-compile", action=DriverName)
+    parser.add_argument("--target", action=TargetOption)
+    parser.add_argument("--DSP-quota", action=NormalOption)
+    parser.add_argument("-o", action=NormalOption)
+    parser.add_argument("input", action=NormalOption)
```

Comment:
It seems that `DriverName`, `TargetOption` and `NormalOption` classes are not defined yet.
Are they going to be implemented later?

I couldn't find `target name` such as `rose` or `sunflower` in `one-codegen_008` test sequence.
Am I right..?

(Actually I couldn't fully understand the sequence of command schema. 😭)

Reply:
> Are they going to be implemented later?

I intentionally didn't implement the body. That classes just do the same with what enum does.

> I couldn't find target name such as rose or sunflower in one-codegen_008 test sequence.

The target is specified in the `one-codegen_008.cfg`;it says `target=one-codegen-008`. So, the test checks the output like below.
```bash
if grep -q "dummy-compile with one-codegen-008 target" "${outputfile}"; then
  echo "${filename_ext} SUCCESS"
  exit 0
fi
```

Reply:
> Actually I couldn't fully understand the sequence of command schema. 

Please refer to the issue for the background on why the command schema was introduced. In a nutshell, it enables backend one-cmds tools such as _one-codegen_ or _one-profile_ to receive arguments in the configuration file according to the backends drivers' interface instead of the `command` key.

Anyway, I'm gonna have to write a document about that but let me elaborate a bit for now.

When the target input is given in the configuration file, _onecc_ would decide the actions based on these priorities.

[precondition] say, for instance, 'rose' target is given.

1. Check if the command schema exists according to the given `rose` target;the path should be like below.
```bash
 # `/usr/share/one/backends/command/${category}/${target}.py`
/usr/share/one/backends/command/codegen/rose.py
```
2. If the schema exists, _onecc_ could interpret the keys accordingly in the `one-codegen` section of the configuration file.
```ini
[backend]
target=rose
[one-codegen]
# $ dummy-compile --target rose -o sample.tvn sample.circle
o=sample.tvn
input=sample.circle
```
3. If the schema doesn't exist, _onecc_ would do as it normally does except one thing;it adds `--target rose` at the beginning of the command.
```ini
[backend]
target=rose
[one-codegen]
backend=dummy
# $ dummy-compile --target rose -o sample.tvn sample.circle
command=-o sample.tvn sample.circle
```

Reply:
@mhs4670go,
Thanks for the detailed description 🙂

One more question, 
So the schema has higher priority than the `command` field in config?

Reply:
@shs-park 

> So the schema has higher priority than the command field in config?

Yes. But, this is only when "target" is given.